docs/contributing.md

```diff
@@ -33,21 +33,81 @@ The next Community Call will be held on October 24th, 2024 at [5:00 PM CEST](htt
 Please use the [project's issue tracker](https://github.com/headlamp-k8s/headlamp/issues) for filing any bugs you find or features
 you think are useful.
 
+### Guidelines for Submitting an Issue
+
+When submitting an issue, follow these guidelines to help maintainers address it efficiently:
+
+- **Search for Existing Issues:** Use the issue tracker to see if your issue already exists. If it does, consider adding your input to the existing issue instead of opening a new one.
+- **Structure Your Issue Description:**  
+  - **Summary:** Briefly explain the issue.  
+  - **Expected Behavior:** What did you expect to happen?  
+  - **Observed Behavior:** What actually happened?  
+  - **Steps to Reproduce:** List clear steps to reproduce the issue.  
+- **Add Relevant Labels:** Use labels such as `bug`, `desktop` to categorize your issue properly.
+
 ### Security issues
 
 For filing security issues that are sensitive and should not be public, please
 send an email to <security@headlamp.dev> .
 
-## Translations
+## Submitting a Pull Request (PR)
 
-If you want to contribute to the internationalization of Headlamp, please refer to the
-dedicated [i18n docs](./development/).
+Follow these steps when submitting a PR to ensure it meets the project’s standards.
 
-### Complex contributions
+### 1. Run Tests and Format Your Code
+
+Navigate to the `frontend` folder and run the following commands:
+
+- `cd frontend`  
+- `npm run test` – Run the test suite.  
+- `npm run tsc` – Check for TypeScript errors.  
+- `npm run format` – Format your code to match the project’s style.
+
+These steps ensure your code is functional, well-typed, and formatted consistently.
+
+---
+
+### 2. Follow Commit Guidelines
+
+Use **atomic commits** to keep each commit focused on a single change. Follow this structure for your commit messages:
+
+`<scope>: <concise description>`
+
+**Examples:**
+
+- `frontend: Fix home button not navigating home`  
+- `backend: Add new API route for Helm chart details`
+
+For more detailed commit practices, see Kinvolk's [Git contribution guidelines](https://github.com/kinvolk/contribution/tree/master/topics/git.md).
+
+---
+
+### 3. Write a Descriptive PR Description
+
+When opening a PR, make sure to:
+
+- **Summarize your changes** and **why** they are needed.  
+- **Link to the related issue**, if applicable.  
```

Comment:
It should add `fixes #ISSUE_NUMBER` , if applicable.

Reply:
done!